clientapi/routing/server_notices.go

```diff
@@ -266,7 +262,36 @@ func SendServerNotice(
 	sendEventDuration.With(prometheus.Labels{"action": "build"}).Observe(float64(timeToGenerateEvent.Milliseconds()))
 	sendEventDuration.With(prometheus.Labels{"action": "submit"}).Observe(float64(timeToSubmitEvent.Milliseconds()))
 
-	return res
+	return res, nil
+}
+
+func getAllUserRooms(ctx context.Context, rsAPI api.RoomserverInternalAPI, userID string) ([]string, error) {
+	allUserRooms := []string{}
+	userRooms := api.QueryRoomsForUserResponse{}
+	if err := rsAPI.QueryRoomsForUser(ctx, &api.QueryRoomsForUserRequest{
+		UserID:         userID,
+		WantMembership: "join",
+	}, &userRooms); err != nil {
+		return nil, err
+	}
+	allUserRooms = append(allUserRooms, userRooms.RoomIDs...)
+	// get invites for specified user
+	if err := rsAPI.QueryRoomsForUser(ctx, &api.QueryRoomsForUserRequest{
```

Comment:
My feeling is that duplicate API calls like this will end up getting expensive on a busy server — we may just want to modify `rsAPI.QueryRoomsForUser` so that if we don't specify `WantMembership`, we just get all rooms regardless of membership.

Reply:
Added a new `MemberShipStateAll` const.